checkout-operator.mk

```diff
@@ -7,7 +7,7 @@ define checkout_operator
 		git clone -b master https://github.com/$(1)/$(2).git $(SOURCE_DIR)/$(2) ;\
 	else \
 		if [[  -z "$${SKIP_GITREFRESH}" ]]; then \
-			cd $(SOURCE_DIR)/$(2) git reset --hard && git pull --force && cd ../.. ;\
+			pushd $(SOURCE_DIR)/$(2) && git reset --hard && git pull --force && popd ;\
```

Comment:
Why use pushd/popd over cd? (I would suggest using the `-n` option to suppress the directory change notice for pushd/popd, but I'm curious why those over cd)

Reply:
Use of cd requires keeping up with the path you've changed to and revert with a bunch of `../` or `-`.  For scripts I generally use pushd because I've nested directory changes in the past which makes `cd -` to revert not work.  It's a pattern that's consistent and doesn't hit human error if the depth of directory structure changes.  +1 for `-n` (I hadn't seen that option before)

Reply:
My guess is that this isn't a strong case to use pushd/popd since cd is so easy to use. It's always going to be 2 depth changes there and 2 depth changes back. But this:

>  It's a pattern that's consistent and doesn't hit human error if the depth of directory structure changes. 

makes it hit home. I like the idea of a consistent _pattern_, even if it's seemingly "silly" here. You'll have a :shipit: from me with the `-n` changes.